libr/asm/asm.c

```diff
@@ -65,9 +65,11 @@ static inline int r_asm_pseudo_org(RAsm *a, char *input) {
 
 // wtf isnt this the same as r_asm_op_set_hex() ??
 static inline int r_asm_pseudo_hex(RAsmOp *op, char *input) {
-	// assume input fits inside op->buf
-	int len = r_hex_str2bin (input, (ut8*)r_strbuf_get (&op->buf));
-	r_asm_op_set_hex (op, r_str_trim_head_tail (input));
+    	ut8 *opBuf = r_asm_op_get_buf (op);
+	// Truncate the input based on the size of the opBuf
+    	input = r_str_newlen (r_str_trim_head_tail (input), sizeof (RStrBuf));
```

Comment:
this is wrong. sizeof(RStrBuf) is not a restriction for anything to care of. also you are reusing a variable that is passed to the function instead of creaing one , so changing its meaning and therefor breaking the logic, and i dont see why you want to truncate that input

Reply:
Okay now after I saw the r_asm_op_set_hex method. I realized the problem was r_hex_str2bin arguments were improperly used. I will revert and fix that.